### ISSUE TITLE
Unusable on SMP server due to memory leaks

### ISSUE THREAD

**Issue by UnixRonin** [SonarSonic/Practical-Logistics-2#26]
Much though I like the functionality this mod provides (it is AWESOME for monitoring the "fallthrough" chest of a RouterReborn-based auto-storage system, for example), I have had to remove it from my 1.10.2 SMP server due to crippling memory leaks.  After about two to three days uptime, the server is completely unresponsive, with CPU usage continually at 200%-225%, and unable to even shut down cleanly.  The problem APPEARS to be Java stack exhaustion rather than Java heap, because there is not generally massive bloating of the server's overall memory usage.

**Comment by SonarSonic** [comment]
@UnixRonin I'm looking into the memory leaks now. I attempted large memory leak fixes for version 2.0.2 was this the reason you were running? If it was what were the main blocks you were using?

**Comment by UnixRonin** [comment]
No, I had to roll back to PracticalLogistics2-1.10.2-2.0.1 and SonarCore-1.10.2-3.2.6 (following the advice of @JasonKillsUS), because 2.0.2/3.2.7 wasn't working at all, I couldn't get displays to connect to and read a data source.

So far I've basically used six parts — nodes, data cables, inventory reader, power reader, full-panel display screen, half-panel display screen.  They work well, when they work, though a few things don't seem to work as expected.  For instance, a loop of data cable connecting a source (say, a furnace) to a screen, with a power reader in one side of the loop and an inventory reader in the other, works.  But rearrange the same parts so that both readers are attached to different faces of the same piece of cable, and only the first reader attached will work.  I was also seeing problems in which after a server restart, some display screens randomly "forget" their configured data sources and have to be reconnected to them.

Side note:
I like the functionality of being able to drop an item into a monitored inventory by just right-clicking the screen with it.  However the item/stack doesn't always go into the correct displayed inventory, if data cables have automatically connected and "networked" several inventories — in that case, items seem to go into the "first available" inventory on the network, with the mod itself deciding which inventory is "first".  That could potentially be an expensive mistake if something that was supposed to go into a furnace goes into a grinder instead (or vice versa).  When I discovered this, it was harmless, because the connected inventories were all a bank of identical furnaces and all that happened was the items went into a different furnace than expected.

**Comment by SonarSonic** [comment]
Okay so for now i'll ignore the memory leaks you mentioned until you update. I'll post a new update that fixes those others soon ish. Then once you've updated let me know if you're still having problems

**Comment by UnixRonin** [comment]
OK, let me know as soon as your new build is up and I'll give it another try.
